pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -70,20 +76,40 @@ def get_profiled_train_dataloader(self, train_dataloader) -> Iterable:
     def prepare_data(self) -> None:
         # on multi-gpu jobs we only want to manipulate (download, etc) on node_rank=0, local_rank=0
         # or in the case where each node needs to do its own manipulation in which case just local_rank=0
-        if self.can_prepare_data():
-            if self.trainer.datamodule is not None:
+        local_rank_zero = self.trainer.local_rank == 0
+        global_rank_zero = self.trainer.local_rank == 0 and self.trainer.node_rank == 0
+
+        datamodule = self.trainer.datamodule
+        lightning_module = self.trainer.lightning_module
+        # handle datamodule prepare data:
+        # check for prepare_data_per_node & datamodule lifecycle properties before calling datamodule.prepare_data
+        if datamodule is not None and not datamodule.has_prepared_data:
+            dm_prepare_data_per_node = datamodule.prepare_data_per_node
+            dm_eq_prepare_data = datamodule.prepare_data_per_node == self.trainer.prepare_data_per_node
+            if (self.trainer.prepare_data_per_node is not None) and not dm_eq_prepare_data:
```

Comment:
nit: extra parenthesis

```suggestion
            if self.trainer.prepare_data_per_node is not None and not dm_eq_prepare_data:
```